### PR TITLE
ELECTRON-483 (Implement a functionality to display snack bar)

### PR DESCRIPTION
**Pull request by KiranNiranjan**
## Description
Displays a snack bar telling user `Press esc to exit full screen` for 3sec whenever the app goes to full screen [ELECTRON-483](https://perzoinc.atlassian.net/browse/ELECTRON-483)

## Approach
How does this change address the problem?
#### Fix:
Displays a snack bar for 3sec whenever the app goes to full screen

## Snack bar
screen shot 2018-06-06 at 8 20 18 pm

## Spectron tests result
[Electron-483 — Spectron.pdf](https://github.com/symphonyoss/SymphonyElectron/files/2076029/Electron-483.Spectron.pdf)

## Unit tests result
[ELECTRON-483 — Unit Tests.pdf](https://github.com/symphonyoss/SymphonyElectron/files/2076031/ELECTRON-483.Unit.Tests.pdf)

## Open Questions if any and Todos
- [ ] Unit-Tests
- [ ] Documentation
- [ ] Automation-Tests

### CODE REVIEW

File: js/preload/preloadMain.js [hunk 1]
```diff
@@ -431,9 +453,19 @@ function createAPI() {
         });
     }
 
+    const throttledKeyDown = throttle(1000, (event) => {
```
Comment (VishwasShashidhar) [inline js/preload/preloadMain.js]: throttling is fine, but I still think this may cause performance issues.

Comment (VishwasShashidhar) [reply js/preload/preloadMain.js]: we'll need to run some tests to confirm that it won't

File: js/snackBar/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    /**
+     * Method that displays a snack bar for 3 sec
+     */
+    showSnackBar() {
+        if (this.snackBar) {
+            this.snackBar.className = "show";
+            this.snackBar.style.display = 'block';
+            setTimeout(() => {
```
Comment (VishwasShashidhar) [inline js/snackBar/index.js]: reason for timeout?

File: js/preload/preloadMain.js [hunk 3]
```diff
@@ -416,6 +421,23 @@ function createAPI() {
         }
     });
 
+    /**
+     * Initializes snack bar
+     */
+    local.ipcRenderer.on('init-snack-bar', () => {
```
Comment (lneir) [inline js/preload/preloadMain.js]: why necessary to have init-snack-bar?  why not just lazy create???  is this some perf issue?

File: js/snackBar/index.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    constructor() {
+        const body = document.getElementsByTagName('body');
+        this.domParser = new DOMParser();
+
+        const snackBar = this.domParser.parseFromString(snackBarContent, 'text/html');
+        this.snackBar = snackBar.getElementById('snackbar');
+
+        if (body && body.length > 0) {
+            body[0].appendChild(this.snackBar);
```
Comment (lneir) [inline js/snackBar/index.js]: why append in constructor?  why not do in show?

File: js/snackBar/index.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    /**
+     * Method that displays a snack bar for 3 sec
+     */
+    showSnackBar() {
+        if (this.snackBar) {
+            this.snackBar.className = "show";
+            this.snackBar.style.display = 'block';
+            setTimeout(() => {
+                this.snackBar.className = this.snackBar.className.replace("show", "");
+                this.snackBar.style.display = 'none';
```
Comment (lneir) [inline js/snackBar/index.js]: this should remove from dom???

File: js/snackBar/index.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+    /**
+     * Method that displays a snack bar for 3 sec
+     */
+    showSnackBar() {
+        if (this.body && this.body.length > 0 && this.snackBar) {
+            this.body[0].appendChild(this.snackBar);
+            this.snackBar.className = "show";
+            setTimeout(() => {
+                this.snackBar.className = this.snackBar.className.replace("show", "");
```
Comment (lneir) [inline js/snackBar/index.js]: no reason for removing show if you are removing from dom.

Comment (KiranNiranjan) [reply js/snackBar/index.js]: Yes, That's correct, removed it 😃

File: js/snackBar/content.js [hunk 7]
```diff
@@ -1,4 +1,10 @@
-const snackBarContent = (`<div id="snackbar">Press <span id="snackbar-esc">esc</span> to exit full screen</div>`);
+const snackBarContent = (
+    `<div id="snackbar">
+       <span data-i18n-text="Press "></span>
+       <span id="snackbar-esc" data-i18n-text="esc"></span>
```
Comment (VikasShashidhar) [inline js/snackBar/content.js]: Should this not be styled a little differently to make it more prominent? Can you talk to Noah about this?

**Pull request merged**